packages/foam-vscode/src/extension.ts

```diff
@@ -126,41 +128,10 @@ async function generateReferenceList(doc: TextDocument): Promise<string[]> {
   const id = dropExtension(basename(filePath));
 
   // @todo fix hack
-  await ws.ready;
+  const foam = await ws.ready;
 
-  // update file in index for future reference
-  // @todo should probably be an update method instead
-  // so we can prune existing references
-  ws.manager.addNoteFromMarkdown(filePath, doc.getText());
-
-  // find note by id
-  const note = ws.manager.getNoteWithLinks(id);
-
-  if (note.linkedNotes.length === 0) {
-    return [];
-  }
-
-  const references = [];
-
-  for (const link of note.linkedNotes) {
-    const relativePath = relative(dirname(filePath), link.absolutePath);
-    if (relativePath) {
-      const relativePathWithoutExtension = dropExtension(relativePath);
-
-      // [wiki-link-text]: wiki-link "Page title"
-      references.push(
-        `[${link.id}]: ${relativePathWithoutExtension} "${link.title}"`
-      );
-    }
-  }
-
-  references.sort()
-
-  // for (const backlink of note.backlinks) {
-  //   references.push(
-  //     `[backlink:${backlink.id}]: ${backlink.filename} "${backlink.title}"`
-  //   );
-  // }
+  const references = uniq(createMarkdownReferences(foam, id)
```

Comment:
now the only thing the extension does is converting the md references for the editor (looking at it even the string creation should go in the core)

Reply:
This is somehow related to the [link reference definition proposal](https://github.com/foambubble/foam/blob/a4277776f8596b12b915231ac2e1d81a67f54f1b/docs/link-reference-definition-improvements.md#improvement-proposal), currently I'm not quite sure in which order it would be beneficial to implement these things

Reply:
yes tbh I am not fully clear on how the link reference would work best, in this case I was reimplementing the existing behavior, but I agree this needs a bit more clarity

Reply:
@riccardoferretti is the purpose of `uniq` here to dedup the list in cases where there were multiple links to the same document? Unless those references are triple equal to each other, this will still leave duplicates as [lodash#uniq uses SameValueZero comparison](https://lodash.com/docs/3.10.1#uniq), which for object compares reference equality, not deep equality.

Reply:
yeah, that's why I am doing the `uniq` on the resulting string, which is kinda of a hack, this should be done at the `createMarkdownReferences` layer (on the object as you hinted at), I was a bit lazy about doing the right comparison. I should really move it there..